File: DependencyInjection/DependencyInjection/_1_ControlFreak/Outbound/XmlTcpOutbound.cs
using DependencyInjection._1_ControlFreak.Interfaces;

namespace DependencyInjection._1_ControlFreak.Outbound
{
  class XmlTcpOutbound
  {
    private readonly TcpSocket _outputSocket;

    public XmlTcpOutbound()
    {
      _outputSocket = new TcpSocket();
    }

    public void Send(AcmeMessage message)
    {
      var outboundMessage = new XmlOutboundMessage();
      message.WriteTo(outboundMessage);
      outboundMessage.SendVia(_outputSocket);
    }
  }
}

File: Java/dependency-injection/_02_control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/Outbound/XmlTcpOutbound.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Outbound;

import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Interfaces.AcmeMessage;

public class XmlTcpOutbound {
  private final TcpSocket outputSocket;

  public XmlTcpOutbound() {
    outputSocket = new TcpSocket();
  }

  public void send(AcmeMessage message) {
    XmlOutboundMessage outboundMessage = new XmlOutboundMessage();
    message.writeTo(outboundMessage);
    outboundMessage.sendVia(outputSocket);
  }
}
